### PR TITLE
Code cleanup: Sorting

### PR DESCRIPTION
**Pull request by jlsnow301**
## About The Pull Request
1. Removes code duplication 
2. Fully documents `sortTim()`
3. Makes a define with default sortTim behavior, straight and to the point for 95% of cases
4. Migrates other sorts into the same file
5. Removes some redundancy where they're reassigning a variable using an in place sorter

For the record, we only use timSort
## Why It's Good For The Game
More documentation, easier to read, uses `length` over `len`, etc
Should be no gameplay effect at all

### CODE REVIEW

File: code/__HELPERS/sorts/helpers.dm [hunk 1]
```diff
@@ -0,0 +1,97 @@
+/// Sorts the list in place with timSort, default settings.
+#define SORT_TIM(to_sort, associative) \
+	if (length(to_sort) < 2) { \
+		return to_sort; \
```
Comment (MrMelbert) [inline code/__HELPERS/sorts/helpers.dm:4]: Dirty return in this macro? Seems like undesired behavior

File: code/__HELPERS/sorts/helpers.dm [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+	sorter.L = to_sort; \
+	sorter.cmp = GLOBAL_PROC_REF(cmp_numeric_asc); \
+	sorter.associative = associative; \
+	sorter.timSort(1, 0); \
+
+
+/// Helper for the sorting procs. Prevents some code duplication. Creates /datum/sort_instance/sorter
+#define CREATE_SORT_INSTANCE(to_sort, cmp, associative, fromIndex, toIndex) \
+	if (length(to_sort) < 2) { \
+		return to_sort; \
```
Comment (MrMelbert) [inline code/__HELPERS/sorts/helpers.dm:19]: Same

File: code/__HELPERS/sorts/helpers.dm (hunk 1 above)
Comment (jlsnow301) [reply code/__HELPERS/sorts/helpers.dm:4]: Wdym dirty here 🤔

Comment (MrMelbert) [reply code/__HELPERS/sorts/helpers.dm:4]: https://github.com/tgstation/tgstation/blob/master/.github/guides/STYLE.md#language---hygienic

File: code/__HELPERS/sorts/helpers.dm (hunk 2 above)
Comment (jlsnow301) [reply code/__HELPERS/sorts/helpers.dm:17]: Does it make sense here, considering it's later undefined? None of the procs continue if its sorted, and making the define hygienic (by not altering from/to) would defeat the purpose of using the define

**Review by ZephyrTFA — APPROVED** [review]

**Pull request merged**